queries/julia/highlights.scm

```diff
@@ -194,124 +142,134 @@
     "end"
   ] @keyword.exception)
 
-(finally_clause
-  "finally" @keyword.exception)
-
 (catch_clause
   "catch" @keyword.exception)
 
+(finally_clause
+  "finally" @keyword.exception)
+
 (for_statement
   [
     "for"
     "end"
   ] @keyword.repeat)
 
+(for_binding
+  "outer" @keyword.repeat)
+
+; comprehensions
+(for_clause
+  "for" @keyword.repeat)
+
+(if_clause
+  "if" @keyword.conditional)
+
 (while_statement
   [
     "while"
     "end"
   ] @keyword.repeat)
 
-(for_clause
-  "for" @keyword.repeat)
-
-(for_binding
-  "outer" @keyword.repeat)
-
 [
   (break_statement)
   (continue_statement)
 ] @keyword.repeat
 
+[
+  "const"
+  "mutable"
+] @keyword.modifier
+
+(function_definition
+  [
+    "function"
+    "end"
+  ] @keyword.function)
+
+(do_clause
+  [
+    "do"
+    "end"
+  ] @keyword.function)
+
+(macro_definition
+  [
+    "macro"
+    "end"
+  ] @keyword)
+
+(return_statement
+  "return" @keyword.return)
+
 (module_definition
   [
     "module"
     "baremodule"
     "end"
   ] @keyword.import)
 
+(export_statement
+  "export" @keyword.import)
+
+(public_statement
+  "public" @keyword.import)
+
 (import_statement
-  [
-    "import"
-    "using"
-  ] @keyword.import)
+  "import" @keyword.import)
+
+(using_statement
+  "using" @keyword.import)
 
 (import_alias
   "as" @keyword.import)
 
-(export_statement
-  "export" @keyword.import)
-
 (selected_import
   ":" @punctuation.delimiter)
 
 (struct_definition
   [
+    "mutable"
     "struct"
     "end"
   ] @keyword.type)
 
-(macro_definition
-  [
-    "macro"
-    "end"
-  ] @keyword)
-
-(function_definition
+(abstract_definition
   [
-    "function"
+    "abstract"
+    "type"
     "end"
-  ] @keyword.function)
+  ] @keyword.type)
 
-(do_clause
+(primitive_definition
   [
-    "do"
+    "primitive"
+    "type"
     "end"
-  ] @keyword.function)
-
-(return_statement
-  "return" @keyword.return)
-
-[
-  "const"
-  "mutable"
-] @keyword.modifier
+  ] @keyword.type)
 
 ; Operators & Punctuation
-[
-  "="
-  "∈"
-  (operator)
-] @operator
+(operator) @operator
 
 (adjoint_expression
   "'" @operator)
 
 (range_expression
   ":" @operator)
 
-((operator) @keyword.operator
-  (#any-of? @keyword.operator "in" "isa"))
-
-(for_binding
-  "in" @keyword.operator)
-
-(where_clause
-  "where" @keyword.operator)
+(arrow_function_expression
+  "->" @operator)
 
-(where_expression
-  "where" @keyword.operator)
+[
+  "."
+  "..."
+  "::"
+] @punctuation.special
 
 [
   ","
-  "."
   ";"
-  "::"
```

Comment:
I feel like this should be left as `@punctuation.delimiter`, looks like most other languages use that highlight for this double colon thing. Unless there was a specific reason

Reply:
Upstream just doesn't like our queries. 🤷